addons/traefik/1.7.x/traefik-11.yaml

```diff
@@ -0,0 +1,130 @@
+---
+apiVersion: kubeaddons.mesosphere.io/v1beta1
+kind: ClusterAddon
+metadata:
+  name: traefik
+  labels:
+    kubeaddons.mesosphere.io/name: traefik
+    kubeaddons.mesosphere.io/provides: ingresscontroller
+  annotations:
+    catalog.kubeaddons.mesosphere.io/addon-revision: "1.7.23-11"
+    appversion.kubeaddons.mesosphere.io/traefik: "1.7.23"
+    endpoint.kubeaddons.mesosphere.io/traefik: "/ops/portal/traefik"
+    docs.kubeaddons.mesosphere.io/traefik: "https://docs.traefik.io/v1.7"
+    values.chart.helm.kubeaddons.mesosphere.io/traefik: "https://raw.githubusercontent.com/mesosphere/charts/00b019ef3610ca8221a8cf283b4d7046a50702c4/staging/traefik/values.yaml"
+spec:
+  kubernetes:
+    minSupportedVersion: v1.15.6
+  requires:
+    - matchLabels:
+        kubeaddons.mesosphere.io/name: cert-manager
+  chartReference:
+    chart: traefik
+    repo: https://mesosphere.github.io/charts/staging
+    version: 1.72.19
+    values: |
+      ---
+      # Configure Traefik for HA.
+      replicas: 2
+      podDisruptionBudget:
+        minAvailable: 1
+      # Distribute pods to tolerate node or zone failure.
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: release
+                  operator: In
+                  values:
+                  - traefik-kubeaddons
+              topologyKey: kubernetes.io/hostname
+          - weight: 1
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: release
+                  operator: In
+                  values:
+                  - traefik-kubeaddons
+              topologyKey: topology.kubernetes.io/zone
```

Comment:
I feel we should not use the `release` label, in this case, because it's controlled and applied _implicitly_ by Helm. (right?) 

This means our correct behavior (HA, upgradability, and tolerance to AZ failure) is predicated on the expected behavior of *an external tool* (i.e. outside the scope of the addon). We should try to minimize implicit dependencies beyond the scope of an addon's definition. 

The chart appears to offer `deployment.podLabels: {}` that would allow us to specify a distinct label here, controlled by the addon definition itself.  I'd suggest creating a distinct pod label, controlled directly in this addon YAML, and using it in the match expression.

Reply:
The release label i think is fine in general, but yeah I see your point where this isnt necessarily managed by us, which would then mean we probably could/would want to add our own labels to make it clear our intention regardless of the tool we are using for deployment.

Reply:
Yeah I wasn't thrilled to use `release` but it was the only workable option among the existing set of labels on Traefik's pods. @samba Good call on `deployment.podLabels`, I'll set something there that we can reference for pod affinity.

Reply:
I went ahead and gave the pods `kubeaddons.mesosphere.io/name` labels in order to avoid inventing my own label: be789c5